xdsl/ir.py

```diff
@@ -641,8 +641,19 @@ def build(
         """Create a new operation using builders."""
         ...
 
-    def replace_operand(self, operand_idx: int, new_operand: SSAValue) -> None:
+    def replace_operand(self, operand: int | SSAValue,
+                        new_operand: SSAValue) -> None:
         """Replace an operand with another operand."""
```

Comment:
nit: maybe comment that it will raise a ValueError if the operand is not an operand of this operation?